core/client/src/main/java/alluxio/client/block/stream/BlockOutStream.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.block.stream;
+
+import alluxio.Constants;
+import alluxio.client.BoundedStream;
+import alluxio.client.Cancelable;
+import alluxio.client.block.BlockStoreContext;
+import alluxio.client.block.BlockWorkerClient;
+import alluxio.client.file.options.OutStreamOptions;
+import alluxio.exception.AlluxioException;
+import alluxio.proto.dataserver.Protocol;
+import alluxio.wire.WorkerNetAddress;
+
+import com.google.common.io.Closer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FilterOutputStream;
+import java.io.IOException;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Provides a stream API to write a block to Alluxio. An instance of this class can be obtained by
+ * calling
+ * {@link alluxio.client.block.AlluxioBlockStore#getOutStream(long, long, OutStreamOptions)}.
+ */
+@NotThreadSafe
+public final class BlockOutStream extends FilterOutputStream implements BoundedStream, Cancelable {
```

Comment:
No, it doesn't. This is only for block out stream. I am not sure why you have this question. So i cannot give a more detailed explanation here. But it doesn't require delegation.